### PR TITLE
[cleanup][Windows] Remove a few compiler warnings and unnecessary global

### PR DESCRIPTION
**Pull request by CrystalP**
## Description

Small code cleanups. Warning removal.

* This is Windows platform code, use the official MS annotation for intentional fallthrough in switch statements, which suppresses the compiler warning.

* Made **TranslateKey()** return early in case of error in **GetKeyboardState()**. This is a little better than current code, which ignores the error code and uses a keystate with undefined content since not initialized and maybe corrupted by the error.

Maybe discarding the key press completely in case of error in **GetKeyboardState()** / inability to retrieve the key modifiers would be better, not sure?

## Motivation and context

Clean the code. Reduce compiler warnings

## How has this been tested?

Builds, quick rt test, but does not change behavior so rt testing is not very helpful.

## What is the effect on users?

None

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/windowing/windows/WinEventsWin32.cpp [hunk 1]
```diff
@@ -167,9 +165,13 @@ static XBMC_keysym *TranslateKey(WPARAM vkey, UINT scancode, XBMC_keysym *keysym
   }
 
   // Attempt to convert the keypress to a UNICODE character
-  GetKeyboardState(keystate);
+  if (!GetKeyboardState(keystate))
```
Comment (thexai) [inline xbmc/windowing/windows/WinEventsWin32.cpp:168]:
```suggestion
  if (FALSE == GetKeyboardState(keystate))
```

GetKeyboardState retuns type BOOL (not bool) and is more correct compare with same type

Comment (CrystalP) [reply xbmc/windowing/windows/WinEventsWin32.cpp]: yes, got confused by MS doc.

**Review by thexai — APPROVED** [review]

**Comment by CrystalP** [comment]
I don't get why sometimes Visual Studio raises warnings or other times not, there was one more in TranslateKeys (static_cast a size_t to UINT)

**Comment by thexai** [comment]
I think you are not talking about build warnings but intellisense / linter text editor warnings. 

https://learn.microsoft.com/en-us/cpp/ide/cpp-linter-overview?view=msvc-170

What fixes this PR are not build warnings (even enabling level 4) but text editor generated hints:

image

Even so it is good to correct them 👍

File: xbmc/windowing/windows/WinEventsWin32.cpp [hunk 2]
```diff
@@ -180,7 +182,8 @@ static XBMC_keysym *TranslateKey(WPARAM vkey, UINT scancode, XBMC_keysym *keysym
       keysym->unicode = static_cast<uint16_t>(vkey - VK_NUMPAD0 + '0');
     }
     else if (ToUnicode(static_cast<UINT>(vkey), scancode, keystate,
-                       reinterpret_cast<LPWSTR>(wchars.data()), wchars.size(), 0) > 0)
+                       reinterpret_cast<LPWSTR>(wchars.data()), static_cast<UINT>(wchars.size()),
```
Comment (thexai) [inline xbmc/windowing/windows/WinEventsWin32.cpp:185]:
```suggestion
                       reinterpret_cast<LPWSTR>(wchars.data()), static_cast<int>(wchars.size()),
```

https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-tounicode

Comment (CrystalP) [reply xbmc/windowing/windows/WinEventsWin32.cpp]: you're right, skipped one in the intellisense.

**Pull request merged**